lib/preload.js

```diff
@@ -91,7 +103,30 @@ const createContainerDisposer = (docker, options) => {
   })
 }
 
-const runContainer = (docker, container) => {
+const pullBuild = (stdout, stdin, dockerPort, imageId) => {
+  // preload.py is running inside the outer Docker container.
+  // We wait until it outputs something on stdout, that means that the inner
+  // Docker is ready.
+  // Then we pull the build and signal the Python script that we're done by
+  // writing on its stdin.
+  stdout.on('data', (data) => {
+    const innerDockerProgress = new dockerProgress.DockerProgress({
+      Promise,
+      host: '0.0.0.0',
+      port: dockerPort
+    })
+    const progress = new visuals.Progress(`Pulling image`)
+    innerDockerProgress.pull(imageId, (event) => {
+      progress.update({ percentage: event.percentage })
+    })
+    .then(() => {
+      // Signal that we're done to the Python script.
+      stdin.write('\n')
```

Comment:
That's the python script's stdin.
Once we're done preloading we signal that it can terminate dockerd and umount the partition(s).
I added the parameters description  in a doc block.